packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -195,38 +196,40 @@ const FormWithRedirect = ({
     };
 
     return (
-        <FormContextProvider value={formContextValue}>
-            <Form
-                key={`${version}_${record?.id || ''}`} // support for refresh button
-                debug={debug}
-                decorators={decorators}
-                destroyOnUnregister={destroyOnUnregister}
-                form={form}
-                initialValues={finalInitialValues}
-                initialValuesEqual={initialValuesEqual}
-                keepDirtyOnReinitialize={keepDirtyOnReinitialize}
-                mutators={finalMutators} // necessary for ArrayInput
-                onSubmit={submit}
-                subscription={subscription} // don't redraw entire form each time one field changes
-                validate={validate}
-                validateOnBlur={validateOnBlur}
-                render={formProps => (
-                    // @ts-ignore Ignored because of a weird error about the active prop
-                    <FormView
-                        {...props}
-                        {...formProps}
-                        key={`${version}_${record?.id || ''}`} // support for refresh button
-                        record={record}
-                        setRedirect={setRedirect}
-                        saving={formProps.submitting || saving}
-                        render={render}
-                        save={save}
-                        warnWhenUnsavedChanges={warnWhenUnsavedChanges}
-                        formRootPathname={formRootPathname}
-                    />
-                )}
-            />
-        </FormContextProvider>
+        <RecordContextProvider value={record}>
```

Comment:
I think in next it should be OptionalRecordContextProvider

Reply:
I'll change it

Reply:
Can you explain to me the introduction of the `OptionalRecordContextProvider` ? To avoid re-renders?